pymc3/backends/base.py

```diff
@@ -552,17 +554,27 @@ def merge_traces(mtraces):
 
     Raises
     ------
-    A ValueError is raised if any traces have overlapping chain numbers.
+    A ValueError is raised if any traces have overlapping chain numbers,
+    or if chains are of different lengths.
 
     Returns
     -------
     A MultiTrace instance with merged chains
     """
+    if len(mtraces) == 0:
+        raise ValueError("Cannot merge an empty set of traces.")
     base_mtrace = mtraces[0]
+    chain_len = len(base_mtrace)
+    # check base trace
+    if any((len(st) != chain_len
+            for _, st in base_mtrace._straces.items())):
```

Comment:
The indent looks off.

Reply:
I think it's actually right, because the second line is inside the iterator comprehension inside the `any`, so it goes in past both the open parens.